spec/models/miq_expression_spec.rb

```diff
@@ -1829,4 +1829,38 @@
       )
     end
   end
+
+  describe "#field_in_sql?" do
+    it "returns false for model.virtualfield" do
+      field = "ManageIQ::Providers::InfraManager::Vm-archived"
+      expression = {"=" => {"field" => field, "value" => "true"}}
+      expect(described_class.new(expression).field_in_sql?(field)).to eq(false)
+    end
+
+    it "returns false for model.association-virtualfield" do
+      field = "ManageIQ::Providers::InfraManager::Vm.storage-v_used_space_percent_of_total"
+      expression = {">=" => {"field" => field, "value" => "50"}}
+      expect(described_class.new(expression).field_in_sql?(field)).to eq(false)
+    end
+
+    it "returns true for model-field" do
+      field = "ManageIQ::Providers::InfraManager::Vm-vendor"
+      expression = {"=" => {"field" => field, "value" => "redhat"}}
+      expect(described_class.new(expression).field_in_sql?(field)).to eq(true)
+    end
+
+    it "returns true for model.association-field" do
+      field = "ManageIQ::Providers::InfraManager::Vm.guest_applications-vendor"
+      expression = {"CONTAINS" => {"field" => field, "value" => "redhat"}}
+      expect(described_class.new(expression).field_in_sql?(field)).to eq(true)
+    end
+
+    it "returns false if column excluded by preprocessing" do
+      field = "EmsClusterPerformance-cpu_usagemhz_rate_average"
+      expression = {">=" => {"field" => field, "value" => "0"}}
+      obj = described_class.new(expression)
+      obj.preprocess_options = {:vim_performance_daily_adhoc => true}
```

Comment:
@imtayadeway I think that this is a victim of the poor implementation of the `vim_performance_daily_adhoc` option. That option is telling it that it's doing some sort of "adhoc" performance processing and as such, it can't used values directly from the DB for some of the columns for evaluation. I think the only thing we can do here is change the description.

Maybe - "returns false if column should be excluded from processing for adhoc performance metrics"